src/cpp/benders/benders_core/BendersBase.cpp

```diff
@@ -249,18 +250,20 @@ void BendersBase::ResetDataPostRelaxation() {
 void BendersBase::check_status(
     const SubProblemDataMap &subproblem_data_map) const {
   if (_data.master_status != SOLVER_STATUS::OPTIMAL) {
-    LOG(INFO) << "Master status is " << _data.master_status << std::endl;
-    throw InvalidSolverStatusException("Master status is " +
-                                       std::to_string(_data.master_status));
+    std::ostringstream msg;
+    auto log_location = LOGLOCATION;
+    msg << "Master status is " + std::to_string(_data.master_status) << std::endl;
+    _logger->display_message(log_location + msg.str());
+    throw InvalidSolverStatusException(msg.str(), log_location);
   }
   for (const auto &[subproblem_name, subproblemData] : subproblem_data_map) {
     if (subproblemData.lpstatus != SOLVER_STATUS::OPTIMAL) {
-      std::stringstream stream;
+      std::ostringstream stream;
+      auto log_location = LOGLOCATION;
       stream << "Subproblem " << subproblem_name << " status is "
-             << subproblemData.lpstatus;
-      LOG(INFO) << stream.str() << std::endl;
-
-      throw InvalidSolverStatusException(stream.str());
+             << subproblemData.lpstatus << std::endl;
+      _logger->display_message(log_location + stream.str());
```

Comment:
It would be nice if `_logger` had an `operator<<`. It would make this code 1 line shorter :)